File: openhpi/branches/3.6.x/baselibs/csharp/openhpi_baselib/OhpiMarshal.cs
/*      -*- c# -*-
 *
 * Copyright (C) 2012, Pigeon Point Systems
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  This
 * file and program are licensed under a BSD style license.  See
 * the Copying file included with the OpenHPI distribution for
 * full licensing terms.
 *
 * Author(s):
 *      Anton Pak <anton.pak@pigeonpoint.com>
 */

using System;
using System.Collections.Generic;


namespace org {
namespace openhpi {


/**********************************************************
 * OHPI Marshal
 *********************************************************/
internal class OhpiMarshal: HpiMarshalGen
{
    /**********************************************************
     * Marshal: For OHPI Simple Data Types
     *********************************************************/
    public void MarshaloHpiHandlerIdT( long x )
    {
        MarshalSaHpiUint32T( x );
    }

    public void MarshaloHpiGlobalParamTypeT( long x )
    {
        MarshalEnum( x );
    }

    /**********************************************************
     * Marshal: For OHPI Structs and Unions
     *********************************************************/
    public void MarshaloHpiHandlerConfigT( oHpiHandlerConfigT x )
    {
        MarshalSaHpiUint8T( x.items.Count );
        foreach ( var kv in x.items ) {
            MarshalByteArray( kv.Key, HpiConst.SAHPI_MAX_TEXT_BUFFER_LENGTH );
            MarshalByteArray( kv.Value, HpiConst.SAHPI_MAX_TEXT_BUFFER_LENGTH );
        }
    }


    /**********************************************************
     * Demarshal: For OHPI Simple Data Types
     *********************************************************/
    public long DemarshaloHpiHandlerIdT()
    {
        return DemarshalSaHpiUint32T();
    }

    public long DemarshaloHpiGlobalParamTypeT()
    {
        return DemarshalEnum();
    }


    /**********************************************************
     * Demarshal: For OHPI Structs and Unions
     *********************************************************/
    public oHpiHandlerInfoT DemarshaloHpiHandlerInfoT()
    {
        oHpiHandlerInfoT x = new oHpiHandlerInfoT();

        x.id = DemarshaloHpiHandlerIdT();
        x.plugin_name = DemarshalByteArray( OhpiConst.MAX_PLUGIN_NAME_LENGTH );
        x.entity_root = DemarshalSaHpiEntityPathT();
        x.load_failed = DemarshalSaHpiInt32T();

        return x;
    }

    public oHpiHandlerConfigT DemarshaloHpiHandlerConfigT()
    {
        oHpiHandlerConfigT x = new oHpiHandlerConfigT();

        x.items = new List< KeyValuePair<byte[], byte[]> >();
        long n = DemarshalSaHpiUint8T();
        for ( int i = 0; i < n; ++i ) {
            byte[] name = DemarshalByteArray( HpiConst.SAHPI_MAX_TEXT_BUFFER_LENGTH );
            byte[] val = DemarshalByteArray( HpiConst.SAHPI_MAX_TEXT_BUFFER_LENGTH );
            x.items.Add( new KeyValuePair<byte[], byte[]>( name, val ) );
        }

        return x;
    }


};


}; // namespace openhpi
}; // namespace org

File: openhpi/tags/pre_ov_rest/baselibs/python/openhpi_baselib/OhpiMarshal.py

#      -*- python -*-
#
# Copyright (C) 2012, Pigeon Point Systems
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  This
# file and program are licensed under a BSD style license.  See
# the Copying file included with the OpenHPI distribution for
# full licensing terms.
#
# Author(s):
#      Anton Pak <anton.pak@pigeonpoint.com>
#

from openhpi_baselib.HpiDataTypes import *
from openhpi_baselib.HpiMarshalGen import HpiMarshalGen
from openhpi_baselib.OhpiDataTypes import *


#**********************************************************
# OHPI Marshal
#**********************************************************
class OhpiMarshal( HpiMarshalGen ):

    #**********************************************************
    # Marshal: For OHPI Simple Data Types
    #**********************************************************
    def marshaloHpiHandlerIdT( self, x ):
        self.marshalSaHpiUint32T( x )

    def marshaloHpiGlobalParamTypeT( self, x ):
        self.marshalEnum( x )


    #**********************************************************
    # Marshal: For OHPI Structs and Unions
    #**********************************************************
    def marshaloHpiHandlerConfigT( self, x ):
        self.marshalSaHpiUint8T( len( x.items ) )
        for ( name, value ) in x.items:
            self.marshalByteArray( name, SAHPI_MAX_TEXT_BUFFER_LENGTH )
            self.marshalByteArray( value, SAHPI_MAX_TEXT_BUFFER_LENGTH )

    def marshaloHpiGlobalParamUnionT( self, x, mod ):
        # TODO
        raise NotImplementedError()

    def marshaloHpiGlobalParamT( self, x ):
        # TODO
        raise NotImplementedError()


    #**********************************************************
    # Demarshal: For OHPI Simple Data Types
    #**********************************************************
    def demarshaloHpiHandlerIdT( self ):
        return self.demarshalSaHpiUint32T()

    def demarshaloHpiGlobalParamTypeT( self ):
        return self.demarshalEnum()


    #**********************************************************
    # Demarshal: For OHPI Structs and Unions
    #**********************************************************
    def demarshaloHpiHandlerInfoT( self ):
        x = oHpiHandlerInfoT()
        x.id = self.demarshaloHpiHandlerIdT()
        x.plugin_name = self.demarshalByteArray( MAX_PLUGIN_NAME_LENGTH )
        x.entity_root = self.demarshalSaHpiEntityPathT()
        x.load_failed = self.demarshalSaHpiInt32T()
        return x

    def demarshaloHpiHandlerConfigT( self ):
        x = oHpiHandlerConfigT()
        x.items = []
        n = self.demarshalSaHpiUint8T()
        for i in range( 0, n ):
            name = self.demarshalByteArray( SAHPI_MAX_TEXT_BUFFER_LENGTH )
            value = self.demarshalByteArray( SAHPI_MAX_TEXT_BUFFER_LENGTH )
            x.items.append( ( name, value ) )
        return x

    def demarshaloHpiGlobalParamUnionT( self, mod ):
        # TODO
        raise NotImplementedError()

    def demarshaloHpiGlobalParamT( self ):
        # TODO
        raise NotImplementedError()


